plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/ParquetTester.java

```diff
@@ -845,25 +852,37 @@ private static long maxPrecision(int numBytes)
     static class ParquetSchemaOptions
     {
         private final boolean singleLevelArray;
+        private final boolean useLegacyDecimalEncoding;
 
-        private ParquetSchemaOptions(boolean singleLevelArray)
+        private ParquetSchemaOptions(boolean singleLevelArray, boolean useLegacyDecimalEncoding)
         {
             this.singleLevelArray = singleLevelArray;
+            this.useLegacyDecimalEncoding = useLegacyDecimalEncoding;
         }
 
         public static ParquetSchemaOptions defaultOptions()
         {
-            return new ParquetSchemaOptions(false);
+            return new ParquetSchemaOptions(false, HIVE_PARQUET_USE_LEGACY_DECIMAL_ENCODING);
         }
 
         public static ParquetSchemaOptions singleLevelArray()
         {
-            return new ParquetSchemaOptions(true);
+            return new ParquetSchemaOptions(true, HIVE_PARQUET_USE_LEGACY_DECIMAL_ENCODING);
+        }
+
+        public static ParquetSchemaOptions integerBackedDecimals()
+        {
+            return new ParquetSchemaOptions(false, false);
```

Comment:
Is there no case where integer-backed decimals and single-level array is needed? Maybe this needs to be turned into a builder.

Reply:
There isn't because single-level array is a non-standard schema that we create through parquet-mr writer and verify that our reader can read, whereas integer backed decimals flag is for the native parquet writer which doesn't produce non-standard schema for arrays.